posts/aws-cognito-and-jhipster-for-the-love-of-oauth-2.0.md

```diff
@@ -0,0 +1,86 @@
+---
+authors:
+- Theo Lebrun
+tags:
+- JHipster
+- AWS
+- AWS Cognito
+- OAuth2
+date: 2020-02-05T14:50:55.000Z
+title: "AWS Cognito and JHipster for the love of OAuth 2.0"
+image: https://raw.githubusercontent.com/Falydoor/blog-usa/cognito-jhipster/images/2020/02/cognito-logo.png
+---
+
+OAuth 2.0 is a stateful security mechanism and OpenID Connect (OIDC) is an authentication layer on top of OAuth 2.0. Spring Security provides excellent OAuth 2.0 and OIDC support, and this is leveraged by JHipster. If you’re not sure what OAuth and OpenID Connect (OIDC) are, please see [What the Heck is OAuth?](https://developer.okta.com/blog/2017/06/21/what-the-heck-is-oauth). JHipster lets you use two OpenID Connect server: Keycloak (default) and Okta. If [Keycloak](https://www.keycloak.org/) is used, you will have to make sure that it is always up and running or it will be impossible to login into your JHipster application. Okta takes care of the hosting part so you don't have to worry on the availability of your OpenID Connect server.
```

Comment:
I would put the objective of the blog at the very beginning to draw the reader in, then follow up with details on each technology.